lib/ansible/modules/cloud/azure/azure_rm_manageddisk.py

```diff
@@ -343,9 +353,12 @@ def exec_module(self, **kwargs):
     def attach(self, vm_name, disk):
         vm = self._get_vm(vm_name)
         # find the lun
-        luns = ([d.lun for d in vm.storage_profile.data_disks]
-                if vm.storage_profile.data_disks else [])
-        lun = max(luns) + 1 if luns else 0
+        if self.lun == 0 or self.lun:
```

Comment:
I think only `self.lun` is enough here. BTY, `self.lun == 0` may fail when the `lun` is `None`.

Reply:
But the lun of the disk starts from 0, it can be 0. The other is that the lun type is an integer and can only be a number. Thanks!
![ttt](https://user-images.githubusercontent.com/37327967/71661167-5dc1b700-2d88-11ea-9015-5516280a0c5a.png)

Reply:
Yes, the `lun` could be 0, but I think there is no necessary to check it here. Maybe you want to check the range of the input?
`isinstance` can check if the input `lun` is a number.

Reply:
Ok.  I will update as your said! Thank you very much!